### DISCUSSION TITLE
请问现在 tx size 的上限是多少？

### DISCUSSION THREAD

**Discussion by doitian** [nervosnetwork/ckb#2839] [General]
如题

**Comment by doitian** [comment]
tx 没有限制, 但是超过 block size limit 不能打包, 建议设置个保守值, 凑个整 2 ** 19 => 524288

via @quake

**Comment by cryptowen** [comment]
帮忙补充一点之前记录的 cycles 限制信息

## cycles 数限制

目前的 ckb mainnet
- 单个块中的 cycles 数总限制： 3_500_000_000
- 除了块的 cycle 限制，单个交易也有限制，现在是 70_000_000

## cycles 数计算手续费

pool 内部会计算 virtual size，交易是按 fee per virtual byte 的优先级打包的。默认最低 rate 是 1 shannon / byte。

1 CKB = 10^8 shannon

```
const DEFAULT_BYTES_PER_CYCLES: f64 = 0.000_17f64;

/// Virtual bytes(aka vbytes) is a concept to unify the size and cycles of a transaction,
/// tx_pool use vbytes to estimate transaction fee rate.
pub(crate) fn get_transaction_virtual_bytes(tx_size: usize, cycles: u64) -> u64 {
    std::cmp::max(
        tx_size as u64,
        (cycles as f64 * DEFAULT_BYTES_PER_CYCLES) as u64,
    )
}
```

## 目前的经验 cycles 消耗

- sUDT 转账：十几万

**Reply by driftluo** [reply]
> 除了块的 cycle 限制，单个交易也有限制，现在是 70_000_000

已经过期了，在下个版本，这个限制将会被开放， pr：https://github.com/nervosnetwork/ckb/pull/2799

**Reply by doitian** [reply]
下个版本指 since v0.100.0
